### ISSUE TITLE
build: standardize tar.gz releases

### ISSUE THREAD

**Issue by itamarjp** [cockroachdb/cockroach#24857]
when I unpack v2.1.0-alpha.20180416.tar.gz I see cockroach-2.1.0-alpha.20180416

in my opinion the tarball should be named, cockroach-2.1.0-alpha.20180416.tar.gz , please think about.

this tarball seems a bit better than older one with src/github.com/cockroachdb/cockroach tree.

**Comment by itamarjp** [comment]
for some reason the stuff inside c-deps in tarball  are empty
./snappy
./googletest
./rocksdb
./jemalloc
./protobuf
./cryptopp

**Comment by danhhz** [comment]
Hey @benesch I don't know much about the source tarball. Is this how the c-deps stuff is expected to look?

**Comment by benesch** [comment]
It sounds pretty broken. The verify-archive TeamCity job keeps a pretty
close eye on the source tarball though so not sure what’s gone wrong. I can
look into it more later today.

**Comment by mberhault** [comment]
I think the tarball in question here is the one from the [github releases](https://github.com/cockroachdb/cockroach/releases) page. We don't control those, everyone should be using the [official images](https://www.cockroachlabs.com/docs/releases/#testing-releases) instead.

**Comment by mberhault** [comment]
Confirmed, the github release is a bit odd:
* the name and link say `v2.1.0-alpha.20180416.tar.gz`, but the actual file downloaded is `cockroach-2.1.0-alpha.20180416.tar.gz` (it extracts in the expected `cockroach-2.1.0-alpha.20180416`)
* all submoduled c-deps subdirectories are empty. pretty sure there's nothing we can do about that.

Unfortunately, github insists on automatically making a "release" for pushed tags without a way to disable it.

Some options here:
* we used to add a link to the release notes page (and download page), maybe we should do that again (and automate it)
* I think there's an API call to delete release assets. This may work.
